database/pebble/db.go

```
@@ -0,0 +1,463 @@
+	}
+
+	if err != nil {
+		return false, updateError(err)
+	}
+	return true, closer.Close()
+}
+
+func (db *Database) Get(key []byte) ([]byte, error) {
+	if db.closed.Get() {
+		return nil, database.ErrClosed
+	}
+
+	data, closer, err := db.db.Get(key)
+	if err != nil {
+		return nil, updateError(err)
+	}
+	ret := slices.Clone(data)
+	return ret, closer.Close()
+}
+
+func (db *Database) Put(key []byte, value []byte) error {
+	// Put causes panic if the db has already been closed
+	if db.closed.Get() {
+		return database.ErrClosed
+	}
+
+	// Use of [pebble.NoSync] here means we don't wait for the [Set] to be
+	// persisted to the WAL before returning. Basic benchmarking indicates that
+	// waiting for the WAL to sync reduces performance by 20%.
+	return updateError(db.db.Set(key, value, pebble.NoSync))
+}
+
+func (db *Database) Delete(key []byte) error {
+	// Delete causes panic if the db has already been closed
+	if db.closed.Get() {
+		return database.ErrClosed
+	}
+
+	return updateError(db.db.Delete(key, pebble.NoSync))
+}
+
+func (db *Database) Compact(start []byte, limit []byte) error {
+	// Pebble Compact causes panic if the db has already been closed.
+	// Avalanche database just return error instead.
+	if db.closed.Get() {
+		return database.ErrClosed
+	}
+
+	if bytes.Equal(start, limit) && limit != nil {
+		// The default compare function of pebble is bytes.Compare.
+		// Use this default compare function since we don't setup
+		// the compare function when we create pebble db.
+		// It's impossible to retrieve this default
+		// compare function from the pebble package since
+		// it's defined in internal.
+		// We use bytes.Equal instead of bytes.Compare
+		// because golangci_lint recommends
+
+		// pebble return error if start and limit are the same,
+		// even if both are nil.
+		// Do nothing just ignore the error if start and limt are the same
+		// and not nil since there is no need to compact and avalanche db
+		// expects a nil return.
```

1. I feel like this comment is a bit too wordy/confusing, I feel like we should just delete it
2. If this if-check only exists because pebble bubble up an error when you have try to compact where `start == limit` can we just bubble up the error instead of silently failing compaction and remove this case? I think all the callers of this just log an error so i think it should be okay